pkg/scheduler/internal/cache/cache.go

```diff
@@ -537,9 +537,7 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 
 	currState, ok := cache.podStates[key]
 	switch {
-	// An assumed pod won't have Delete/Remove event. It needs to have Add event
-	// before Remove event, in which case the state would change from Assumed to Added.
-	case ok && !cache.assumedPods.Has(key):
+	case ok:
```

Comment:
Can we have this UTed? (i.e., pod deletions w/ and w/o the pod key in assumedPods.)